packages/client/src/__tests__/integration/happy/interactive-transactions-postgres/test.ts

```diff
@@ -0,0 +1,473 @@
+              Transaction API error: Transaction already closed: Transaction is no longer valid. Last state: 'Committed'.
+          `)
+
+    const users = await prisma.user.findMany()
+
+    expect(users.length).toBe(0)
+  })
+
+  /**
+   * Batching should work with using the interactive transaction logic
+   */
+  test('batching', async () => {
+    await prisma.$transaction([
+      prisma.user.create({
+        data: {
+          email: 'user_1@website.com',
+        },
+      }),
+      prisma.user.create({
+        data: {
+          email: 'user_2@website.com',
+        },
+      }),
+    ])
+
+    const users = await prisma.user.findMany()
+
+    expect(users.length).toBe(2)
+  })
+
+  /**
+   * A bad batch should rollback using the interactive transaction logic
+   * // TODO: skipped because output differs from binary to library
+   */
+  test.skip('batching rollback', async () => {
+    const result = prisma.$transaction([
+      prisma.user.create({
+        data: {
+          email: 'user_1@website.com',
+        },
+      }),
+      prisma.user.create({
+        data: {
+          email: 'user_1@website.com',
+        },
+      }),
+    ])
+
+    await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(`
+
+                                    Invalid \`prisma.user.create()\` invocation in
+                                    /client/src/__tests__/integration/happy/interactive-transactions-concurrent-postgres/test.ts:0:0
+
+                                      268  */
+                                      269 test('batching rollback', async () => {
+                                      270   const result = prisma.$transaction([
+                                    → 271     prisma.user.create(
+                                      Unique constraint failed on the fields: (\`email\`)
+                              `)
+
+    const users = await prisma.user.findMany()
+
+    expect(users.length).toBe(0)
+  })
+
+  /**
+   * A bad batch should rollback using the interactive transaction logic
+   */
+  test('batching raw rollback', async () => {
+    await prisma.user.create({
+      data: {
+        email: 'user_1@website.com',
+      },
+    })
+
+    const result = prisma.$transaction([
+      prisma.$executeRaw`INSERT INTO User (id, email) VALUES ("2", "user_2@website.com")`,
+      prisma.$queryRaw`DELETE FROM User`,
+      prisma.$executeRaw`INSERT INTO User (id, email) VALUES ("1", "user_1@website.com")`,
+      prisma.$executeRaw`INSERT INTO User (id, email) VALUES ("1", "user_1@website.com")`,
+    ])
+
+    await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(`
+
+                                                                                                Invalid \`prisma.executeRaw()\` invocation:
+
+
+                                                                                                  Raw query failed. Code: \`42601\`. Message: \`db error: ERROR: syntax error at or near "User"\`
```

Comment:
Is this the right error? I would've expected this to check that a unique constraint failed after the last `INSERT` statement. Was this test copied from the SQLite tests? Maybe the SQL statements need to be slightly adapted.

Reply:
Good catch, thanks!